frontend/src/metabase/query_builder/reducers.js

```diff
@@ -204,6 +204,12 @@ export const uiControls = handleActions(
         initialChartSetting: payload,
       }),
     },
+    [CANCEL_QUESTION_CHANGES]: {
+      next: state => ({
+        ...state,
+        isModifiedFromNotebook: false,
```

Comment:
Without it `getShouldShowUnsavedChangesWarning` would return `true` after canceling (via "Cancel" button, but not browser's back button) editing query of a model/metric, which would result in two consecutive unsaved changes warnings.